chainer/optimizer_hooks/data_parallel.py

```diff
@@ -0,0 +1,53 @@
+import chainer
+import chainer.links.model
+
+
+class DataParallelOptimizerCumulateGradientsHook(object):
+    """
+    A hook which sums up all replication's gradients in a
+    DataParallel-Scenario
+    """
+
+    name = "DataParallelCumulateGradients"
+    call_for_each_param = False
+    timing = 'pre'
+
+    def __call__(self, optimizer: chainer.Optimizer):
```

Comment:
I guess you should remove the type annotation as `chainer` supports Python 2.7

https://docs.chainer.org/en/stable/install.html

Plus it will also fix the error.